src/framework/preferences/css/stylus/utils/Themes.styl

```
@@ -1,250 +1,374 @@
-    for themeName in contrastThemes {
-        fColor = contrastThemes[themeName].foregroundColor;
-        bColor = contrastThemes[themeName].backgroundColor;
-
-        {themeName} {
-            background-image: none !important;
-            get-colors-for-separatedPanelPrefsEditor(fColor, bColor);
-
-            .fl-prefsEditor-separatedPanel {
-                .fl-panelBar, .fl-prefsEditor-reset {
-                    get-colors-for-separatedPanelPrefsEditor(fColor, bColor);
-                }
-                .fl-prefsEditor-buttons { //FLUID-4991: The shadow effect around the hide/show preferences bug fix.
-                    box-shadow:none;
-                }
-            }
+// For the general theme styling that applies to html elements and infusion components
+build-themes-Enactors() {
+    [class*="fl-theme"]:not(.fl-theme-prefsEditor-default) {
+        &, *, .fl-pre view-theme&,
+        div, input,
+        h1, h2, h3, h4, h5, h6 {
+            color: var(--fl-fgColor) !important;
+            background-color: var(--fl-bgColor) !important;
+            border-top-color: var(--fl-fgColor) !important;
+            border-right-color: var(--fl-fgColor) !important;
+            border-bottom-color: var(--fl-fgColor) !important;
+            border-left-color: var(--fl-fgColor) !important;
         }
-    }
-}
 
-// For the separated panel iframe
-build-themes-separatedPanelPrefsEditorFrame(contrastThemes) {
-    for themeName in contrastThemes {
-        .fl-prefsEditor-separatedPanel{themeName} {
-            background-image: none;
+        &:disabled,
+        :disabled,
+        :disabled legend,
+        input[type="button"]:disabled,
+        input[type="button"]:disabled *,
+        input[type="reset"]:disabled,
+        input[type="reset"]:disabled *,
+        input[type="submit"]:disabled,
+        input[type="submit"]:disabled *  {
+            color: var(--fl-disabledColor) !important;
         }
-    }
-}
 
-// For the general theme styling that applies to html elements and infusion components
-build-themes-Enactors(contrastThemes) {
-    for themeName in contrastThemes {
-        fColor = contrastThemes[themeName].foregroundColor;
-        bColor = contrastThemes[themeName].backgroundColor;
-        linkColor = contrastThemes[themeName].linkColor || fColor;
-        buttonFColor = contrastThemes[themeName].buttonForegroundColor || fColor;
-        buttonBColor = contrastThemes[themeName].buttonBackgroundColor || bColor;
-
-        disabledColor = contrastThemes[themeName].disabledColor;
-        selectedFColor = contrastThemes[themeName].selectedForegroundColor;
-        selectedBColor = contrastThemes[themeName].selectedBackgroundColor;
-
-
-        {themeName} {
-            // General Styling
-            &, *, .fl-preview-theme&,
+        &::selection,
+        ::selection {
+            color: var(--fl-selectedFColor) !important;
+            background-color: var(--fl-selectedBColor) !important;
```

Is there a typo here, should it be:
```suggestion
            background-color: var(--fl-selectedBgColor) !important;
```